build.sbt

```diff
@@ -44,6 +51,7 @@ val munit = Def.setting("org.scalameta" %%% "munit" % "1.0.0-M7")
 val core = crossProject(JVMPlatform, NativePlatform).crossType(CrossType.Pure).settings(
   name := "mima-core",
   crossScalaVersions ++= Seq(scala213, scala3),
+  scalacOptions ++= compilerOptions(scalaVersion.value),
```

Comment:
Why not have it build-wide?  We set the scala version build-wide.

Reply:
Because I tried it that way and it didn't work. In this build (har har har), `scalaVersion` isn't controlled at the `ThisBuild` level, it's controlled at the per-project level. So if we want to make `scalacOptions` vary depending on `scalaVersion`, we must also set `scalacOptions` in a per-project way.

Unless I'm missing something, but I did try the `ThisBuild` way first.